### ISSUE TITLE
Ошибка в dev версии

### ISSUE THREAD

**Issue by lyolic** [shvchk/padavan-wireguard-client#9]
Если запустит и остановить скрипт, то появляются следующие ошибки во время запуска:
/etc/storage/wireguard # /etc/storage/wireguard/client.sh start
RTNETLINK answers: No such process
RTNETLINK answers: No such file or directory
RTNETLINK answers: No such file or directory
iptables: Bad rule (does a matching rule exist in that chain?).
iptables: No chain/target/match by that name.
iptables: No chain/target/match by that name.
/etc/storage/wireguard # /etc/storage/wireguard/client.sh stop
RTNETLINK answers: No such file or directory
/etc/storage/wireguard #

**Comment by endlocal** [comment]
Аналогично. При первом запуске все прошло отлично, соединение успешно создалось и все работало. Теперь при запуске аналогичные ошибки и полностью падают все соединения

**Comment by shvchk** [comment]
Мне теперь сложно проверять, т.к. не использую падаван. Для отладки я бы сначала заменил `set -euo pipefail` в начале скрипта на `set -euxo pipefail`, т.е. добавил `x` — это покажет все выполняемые скриптом действия и таким образом будет легче понять, какое действие приводит к ошибке. Скиньте результат, а также источник и версию прошивки, я посмотрю. Не гарантирую, но может быть и проверю на своём резервном роутере, когда/если будет время.

**Comment by endlocal** [comment]
> Мне теперь сложно проверять, т.к. не использую падаван. Для отладки я бы сначала заменил `set -euo pipefail` в начале скрипта на `set -euxo pipefail`, т.е. добавил бы `x` — это покажет все выполняемые скриптом действия и таким образом будет легче понять, какое действие приводит к ошибке. Скиньте результат, а также источник и версию прошивки, я посмотрю. Не гарантирую, но может быть и проверю на своём резервном роутере, когда/если будет время.

Оказалось, что во флеше не все файлы записались, а я не заметил. Перезалил все и заработало, спасибо огромное.

Может подскажите, такой момент не соображу. Если на стороне сервера допускаю работу с NAT, то все прекрасно - весь трафик уходит на сервер. Но мне просто нужно связать разные локальные сети.
Конфиг, который я запускаю на windows-клиенте работает корректно, использует только ту сеть, которую я прописал для связи "AllowedIPs = 192.168.23.0/24. Этот же конфиг на роутере отлично работает и пускает меня в нужную локальную сеть, но при этом, вероятно, пытается весь трафик пустить в этот же туннель, хотя в настройках указана только одна подсеть ENDPOINT_ALLOWED_IP="192.168.23.0/24". Почему так может происходить? При этом лог забит сообщениями No peer has allowed IPs matching

**Comment by shvchk** [comment]
А, вы используете более старую версию, из master ветки. У этого ишью в названии указана dev версия (переработанная, более функциональная и простая в использовании), так что изначально я исходил из этого.

В dev весрии такого поведения быть не должно, она по AllowedIPs как раз понимает, когда направлять весь трафик, а когда нет. В master версии придётся редактировать скрипт: попробуйте в `traffic_rules.sh` закомментировать строку 74: `ip rule $ip_action to all table 51 pref 40`

**Comment by endlocal** [comment]
> А, вы используете более старую версию, из master ветки. У этого ишью в названии указана dev версия (переработанная, более функциональная и простая в использовании), так что изначально я исходил из этого.
> 
> В dev версии такого поведения быть не должно, она по AllowedIPs как раз понимает, когда направлять весь трафик, а когда нет. В master версии придётся редактировать скрипт: попробуйте в `traffic_rules.sh` закомментировать строку 74: `ip rule $ip_action to all table 51 pref 40` (не забудьте затем сохранить изменения, чтобы не потерялись после перезагрузки: `mtd_storage.sh save`)

Да, действительно. Все перемешал я в одну кучу я. Dev успешно запустилась, все работает как часы. Спасибо за помощь!
